infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -600,6 +610,26 @@ public void accept(ProbeResult probeResult) {
     }
   }
 
+  /** Listens container's events and terminates runtime if unrecoverable event occurs. */
+  public class UnrecoverableEventHanler implements ContainerEventHandler {
+
+    @Override
+    public void handle(ContainerEvent event) {
+      String reason = event.getReason();
+      String message = event.getMessage();
+      if (isUnrecoverable(reason)) {
+        LOG.error("Unrecoverable event occured {}, {}", reason, message);
+        eventPublisher.sendRuntimeStoppedEvent(
+            format("Unrecoverable event occured: '%s', '%s'", reason, message),
+            getContext().getIdentity());
+      }
+    }
+
+    private boolean isUnrecoverable(String reason) {
+      return reason.equals("Failed Mount") || reason.equals("Failed Scheduling");
```

Comment:
Maybe it makes sense to make these values configurable?

Reply:
+1

Reply:
@sleshchenko do you mean comma separated property ?

Reply:
I think yes.

Reply:
done